### PR TITLE
MRG BUG enable use of on demand instances

### PR DESCRIPTION
**Pull request by maikia**
The use of on demand instances on AWS is no longer working.

This PR corrects it and adds a test

### CODE REVIEW

**Review by agramfort — APPROVED** [review]

File: ramp-engine/ramp_engine/tests/test_aws.py [hunk 1]
```diff
@@ -33,6 +35,25 @@ def add_empty_dir(dir_name):
         os.mkdir(dir_name)
 
 
+@pytest.mark.parametrize(
+    "use_spot_instance",
+    [None, True, False]
+    )
+@mock.patch("boto3.session.Session")
+def test_launch_ec2_instances(boto_session_cls, use_spot_instance):
+    ''' 'use_spot_instance' settings run through if True or False'''
```
Comment (lucyleeow) [inline ramp-engine/ramp_engine/tests/test_aws.py:44]:
```suggestion
    '''Check 'use_spot_instance' config with None, True and False.'''
```

**Review by lucyleeow — COMMENTED** [review]
LGTM!

File: ramp-engine/ramp_engine/aws/api.py [hunk 2]
```diff
@@ -144,6 +143,7 @@ def launch_ec2_instances(config, nb=1):
     sess = _get_boto_session(config)
     client = sess.client('ec2')
     resource = sess.resource('ec2')
+    on_demand = False
```
Comment (lucyleeow) [inline ramp-engine/ramp_engine/aws/api.py:146]: To make this clearer, I think we should change this var name to `switch_to_on_demand`

**Comment by maikia** [comment]
thanks @lucyleeow and @agramfort

**Pull request merged**